feat(sdk): Add support for insecure secrets for when running non-secure mode (#285)
edgexfoundry/app-functions-sdk-go: README.md

> *NOTE: Changing Writable.Pipeline.ExecutionOrder will invalidate all currently stored data and result in it all being removed from the database on the next retry.* This is because the position of the export function can no longer be guaranteed and no way to ensure it is properly executed on the retry.

### Secrets

#### Getting Secrets

Application Services can retrieve their secrets from the underlying secret store using the [GetSecrets()](#.GetSecrets()) API in the SDK. 

##### Vault Secrets

If in secure mode, the secrets are retrieved from Vault based on the SecretStore configuration values. If tokenfile is present, it will override the Authentication.AuthToken value. The path parameter in the GetSecrets() API call is appended to the SecretStore.Path configuration and used as the secret's location in Vault. 

##### Insecure Secrets

> When security is disabled, the secrets are retrieved from the *Writable.InsecureSecrets* section of the configuration file. Insecure secrets and their paths can be configured as below.

Reviewer: ```suggestion
When running in insecure mode, the secrets are retrieved from the *Writable.InsecureSecrets* section of the configuration file. Insecure secrets and their paths can be configured as below.
```
 minor modification of sentence suggesting.

> *NOTE: Changing Writable.Pipeline.ExecutionOrder will invalidate all currently stored data and result in it all being removed from the database on the next retry.* This is because the position of the export function can no longer be guaranteed and no way to ensure it is properly executed on the retry.

### Secrets

#### Getting Secrets

Application Services can retrieve their secrets from the underlying secret store using the [GetSecrets()](#.GetSecrets()) API in the SDK. 

##### Vault Secrets

> If in secure mode, the secrets are retrieved from Vault based on the SecretStore configuration values. If tokenfile is present, it will override the Authentication.AuthToken value. The path parameter in the GetSecrets() API call is appended to the SecretStore.Path configuration and used as the secret's location in Vault. 

Reviewer 2: `If tokenfile is present, it will override the Authentication.AuthToken value` seems out of place. I.e. belongs in a section on configuring SecretStore, which we don't have and is beyond this PR. Probably should add an issue for this. For now just remove.
